File: packages/core/dto/configmap.go
package dto

type ConfigMapRef struct {
	Namespace string `json:"namespace"`
	Name      string `json:"name"`
}

type ConfigMap struct {
	Name          string
	Namespace     string
	Keys          []string
	Age           string
	CreatedAt     string
	Labels        map[string]string
	Annotations   map[string]string
	ManagedFields []ManagedField
	Data          map[string]string
}

File: packages/core/frontend/src/types/resources/configmap.ts
import type { ManagedField } from "./shared";

export interface ConfigMap {
  Name: string;
  Namespace: string;
  Keys: string[];
  Age: string;
  CreatedAt: string;
  Labels: Record<string, string>;
  Annotations: Record<string, string>;
  ManagedFields: ManagedField[];
  Data: Record<string, string>;
}
